src/main/java/com/gmailatgauru/dogra/exceltoexcelparsingwebapp/ReadXlsx.java

```diff
@@ -0,0 +1,51 @@
+package com.gmailatgauru.dogra.exceltoexcelparsingwebapp;
+
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+public class ReadXlsx {
+
+    public void parse(String filePath) throws IOException {
+        FileInputStream file = readFile(filePath);
+        Workbook workbook = getWorkbook(file);
+        Sheet firstSheet = getFirstSheet(workbook);
+        printDataFromSheet(firstSheet); // for testing purpose
+    }
+
+    private void printDataFromSheet(Sheet firstSheet) {
+        System.out.println(firstSheet.getSheetName());
+    }
+
+    public FileInputStream readFile(String filePath) throws FileNotFoundException {
```

Comment:
I don't see where and when you actually close your file input stream. Am I missing something? Can you read the file and close the stream or you need the stream to remain open for some reason?

Reply:
I'll amend